ggml/src/ggml-cpu/amx/mmq.cpp

```diff
@@ -2379,7 +2400,7 @@ void ggml_backend_amx_mul_mat(ggml_backend_amx_context * ctx, struct ggml_tensor
         const int MB = div_up(M, BLOCK_M);
         const int NB = div_up(N, BLOCK_N);
 
-        parallel_for(n_threads, MB * NB, [&](int begin, int end) {
+        parallel_for_ggml(params, MB * NB, [&](int begin, int end) {
```

Comment:
Is this fp16 matmul faster than LLAMAFILE fp16 sgemm? 
if not, now this backend is in the CPU backend it may not be needed.

Reply:
Strange, it doesn't use AVX and yet it looks like the same way of doing things as with tinyblas. I'll have to look at this more closely :sunglasses:

Reply:
https://github.com/ggerganov/llama.cpp/blob/3420909dffa50e70660524797a1e715a717684d2/ggml/src/ggml-cpu/llamafile/sgemm.cpp#L421-L445

Reply:
I did not put AMX in the old patch because the 6th gen Xeon is not released at that moment. Since it is on sale now, i think it is good to add `amx-f16` to the gemm.